### PR TITLE
Add tests

### LINKED ISSUE #2: Unit tests?
This seems like a really promising vim library. I'd like to see more projects migrating to rely on this, if it's robust enough. I'd love to be able to move the `if has('nvim')` clauses out of my vimrc.

That being said, are you planning on adding unit tests? I don't have much time on my hands, but I may be able to help out if you can get something started. I'm not sure if you have a favorite vim testing framework yet, but I have some experience using [vroom](https://github.com/google/vroom).

### PR DESCRIPTION
**Pull request by mattn**
Closes #2

### CODE REVIEW

File: test/async.vimspec [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      let seconds = reltimefloat(reltime(start))
+      call async#job#stop(job)
+      Assert seconds > 2
+    End
+  End
+
+  Describe async#job#send
+    It can send input text to the job
+      let s:output = []
+      let job = async#job#start('cat', {'on_stdout': function('s:on_stdout')})
```
Comment (prabirshrestha) [inline test/async.vimspec]: Can we instead do this? That way we can avoid global var.

```vim

function! s:on_stdout(ctx, id, data, event) abort
    call add(a:ctx,output, join(a:data, ''))
endfunction

let ctx = { 'output': [] }
let job = async#job#start('cat', {'on_stdout': function('s:on_stdout', [ctx])})
```
If possible I would prefer lambdas and then use `job#wait` in the next line after job.

File: test/async.vimspec [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  Describe async#job#stop
+    It can stop the job specified
+      let job = async#job#start('bash -c "sleep 2 && touch i-love-vim"', {})
+      sleep 3
+      call async#job#stop(job)
+      Assert filereadable('i-love-vim')
+      call delete('i-love-vim')
+
+      let job = async#job#start('bash -c "sleep 2 && touch i-love-vim"', {})
+      sleep 1
```
Comment (prabirshrestha) [inline test/async.vimspec:23]: for now ok. But would be good to have the same consistent sleep time.

**Review by prabirshrestha — CHANGES_REQUESTED** [review]

File: test/async.vimspec (hunk 1 above)
Comment (mattn) [reply test/async.vimspec]: I'm not familier to neovim but neovim already support parcial arguments?

**Review by prabirshrestha — APPROVED** [review]

**Comment by prabirshrestha** [comment]
Approved. Feel free to merge.

**Pull request merged**